src/Adapter/Title/CommandHandler/BulkDeleteTitleHandler.php

```diff
@@ -28,28 +28,23 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Title\CommandHandler;
 
-use Gender;
+use PrestaShop\PrestaShop\Adapter\Title\AbstractTitleHandler;
 use PrestaShop\PrestaShop\Core\Domain\Title\Command\BulkDeleteTitleCommand;
 use PrestaShop\PrestaShop\Core\Domain\Title\CommandHandler\BulkDeleteTitleHandlerInterface;
 use PrestaShop\PrestaShop\Core\Domain\Title\Exception\DeleteTitleException;
-use PrestaShop\PrestaShop\Core\Domain\Title\Exception\TitleNotFoundException;
 
 /**
  * Handles command that bulk delete titles
  */
-class BulkDeleteTitleHandler implements BulkDeleteTitleHandlerInterface
+class BulkDeleteTitleHandler extends AbstractTitleHandler implements BulkDeleteTitleHandlerInterface
 {
     /**
      * {@inheritdoc}
      */
     public function handle(BulkDeleteTitleCommand $command): void
     {
         foreach ($command->getTitleIds() as $titleId) {
-            $title = new Gender($titleId->getValue());
-
-            if (0 >= $title->id) {
-                throw new TitleNotFoundException(sprintf('Unable to find title with id "%d" for deletion', $titleId->getValue()));
-            }
+            $title = $this->titleRepository->get($titleId);
```

Comment:
```suggestion
            $this->titleRepository->delete($this->titleRepository->get($titleId));
```

Reply:
Only this one line should cover it all even the exception throwing

Reply:
Fixed